packages/@blockprotocol/graph/src/stdlib/subgraph/builder.ts

```diff
@@ -0,0 +1,68 @@
+import { addEntitiesToSubgraphByMutation } from "../../internal/mutate-subgraph.js";
+import {
+  Entity,
+  EntityEditionId,
+  GraphResolveDepths,
+  Subgraph,
+} from "../../types.js";
+
+/**
+ * Builds a Subgraph from a given set of entities, which may include 'link entities' –
+ * i.e. entities that represent relationships between other entities – or other entities.
+ *
+ * The set of entities should represent the result of a query on a graph,
+ * and the 'roots' and 'depths' for that query provided along with the data.
+ *
+ * Unbounded queries can currently only be represented via very high values for depths.
```

Comment:
I think we should just say that query depths are bounded by a maximum of 255. (We should encode this in the spec, or change our backend implementation to allow for greater depths)